services/checks/checks.go

```diff
@@ -419,6 +420,118 @@ func (s *Service) EnableChecks(checkNames []string) error {
 	return nil
 }
 
+// ChangeInterval changes a check's interval to the value received from the UI.
+func (s *Service) ChangeInterval(params map[string]managementpb.SecurityCheckInterval) error {
+	checks := s.GetAllChecks()
+	if len(checks) == 0 {
+		return errors.New("no checks loaded")
+	}
+
+	checkMap := make(map[string]check.Check, len(checks))
+	for _, ch := range checks {
+		checkMap[ch.Name] = ch
+	}
+
+	checkStateMap, err := models.FindCheckStates(s.db.Querier)
```

Comment:
Not sure, bet maybe it makes sense to not load all states. We know check names that we want to update.

Reply:
agreed it was done after this review comment https://github.com/percona/pmm-managed/pull/712#discussion_r606407886 

right now we just change interval for a single check so we don't need to load all but in case this feature is turned into a bulk action on the UI in the future, we won't have to change the backend and it will still work

Reply:
Yeah, but it was requested in another place, not here.

Reply:
you are right I made the change in the wrong place. will revert this and make that change in collect checks

Reply:
done, now it loads all the checkstates in `collectChecks` and `ChangeInterval` is back to the way it was retrieving one state at a time using name